### PR TITLE
Switch to kotlin

### PR DESCRIPTION
**Pull request by hwbllmnn**
Converts all source files to Kotlin.

@terrestris/devs Please review.

### CODE REVIEW

**Review by dnlkoch — APPROVED** [review]

File: src/main/kotlin/de/terrestris/maven/i18n/I18nCombineMojo.kt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    private fun combineJsonFiles(node: ObjectNode, dir: File) {
+        val files = dir.listFiles() ?: return
+        for (file in files) {
+            if (file.isDirectory) {
+                combineJsonFiles(node, file)
+            }
+            if (file.name.endsWith(".i18n.json")) {
+                val current = mapper.readTree(file) as ObjectNode
+                val name = file.name.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()[0]
```
Comment (simonseyock) [inline src/main/kotlin/de/terrestris/maven/i18n/I18nCombineMojo.kt:73]:
```suggestion
                val name = file.name.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.first()
```

File: src/main/kotlin/de/terrestris/maven/i18n/I18nFillMojo.kt [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            if (file.name.endsWith(".i18n.json")) {
+                val contents = mapper.readTree(file) as ObjectNode
+                val source = contents.get(sourceLanguage) as ObjectNode
+                var target = contents.get(targetLanguage) as ObjectNode?
+                if (target == null) {
+                    target = factory.objectNode()
+                    contents.set(targetLanguage, target)
+                }
+                for (key in source.fieldNames()) {
+                    if (!target!!.has(key)) {
```
Comment (simonseyock) [inline src/main/kotlin/de/terrestris/maven/i18n/I18nFillMojo.kt:82]:
```suggestion
                    if (!target.has(key)) {
```
if `factory.objectNode()` does not return `null` this should work

File: src/main/kotlin/de/terrestris/maven/i18n/I18nSplitMojo.kt [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                splitJsonFile(node, file)
+            }
+            if (file.name.endsWith(".i18n.json")) {
+                val contents = mapper.readTree(file) as ObjectNode
+                var current: ObjectNode? = contents.get(language) as ObjectNode?
+                if (current == null) {
+                    current = factory.objectNode()
+                    contents.set(language, current)
+                }
+                val name = file.name.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()[0]
```
Comment (simonseyock) [inline src/main/kotlin/de/terrestris/maven/i18n/I18nSplitMojo.kt:76]:
```suggestion
                val name = file.name.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.first()
```

File: src/main/kotlin/de/terrestris/maven/i18n/I18nSplitMojo.kt [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                val newValues = node.get(name) as ObjectNode
+                for (key in newValues.fieldNames()) {
+                    current!!.set(key, newValues.get(key))
```
Comment (simonseyock) [inline src/main/kotlin/de/terrestris/maven/i18n/I18nSplitMojo.kt:79]:
```suggestion
                    current.set(key, newValues.get(key))
```
see above

**Review by simonseyock — COMMENTED** [review]
awsome! just some nitpicks :)

File: src/main/kotlin/de/terrestris/maven/i18n/I18nFillMojo.kt (hunk 2 above)
Comment (hwbllmnn) [reply src/main/kotlin/de/terrestris/maven/i18n/I18nFillMojo.kt:82]: This doesn't work, the compiler complains here.

File: src/main/kotlin/de/terrestris/maven/i18n/I18nSplitMojo.kt (hunk 4 above)
Comment (hwbllmnn) [reply src/main/kotlin/de/terrestris/maven/i18n/I18nSplitMojo.kt:79]: Same here, unfortunately

File: src/main/kotlin/de/terrestris/maven/i18n/I18nCombineMojo.kt (hunk 1 above)
Comment (hwbllmnn) [reply src/main/kotlin/de/terrestris/maven/i18n/I18nCombineMojo.kt]: Fixed in ae11aa3

File: src/main/kotlin/de/terrestris/maven/i18n/I18nSplitMojo.kt (hunk 3 above)
Comment (hwbllmnn) [reply src/main/kotlin/de/terrestris/maven/i18n/I18nSplitMojo.kt]: Fixed in ae11aa3

**Pull request merged**